src/cli/analytics.rs

```diff
@@ -77,51 +85,103 @@ impl Command for Analytics {
+            let (start, _) = result_string.split_at(result_string.len() - 1);
+            let with_cluster = format!("{}, \"cluster\": \"{}\" }}", start, cluster);
+            let json_object = serde_json::from_str::<serde_json::Value>(&with_cluster).unwrap();
+            Some(convert_json_value_to_nu_value(&json_object, self.span).unwrap())
+        } else {
+            None
+        }
+    }
+}
+
 fn run(
     state: Arc<Mutex<State>>,
     engine_state: &EngineState,
     stack: &mut Stack,
     call: &Call,
     _input: PipelineData,
 ) -> Result<PipelineData, ShellError> {
-    let ctrl_c = engine_state.ctrlc.as_ref().unwrap().clone();
-    let statement: String = call.req(engine_state, stack, 0)?;
     let span = call.head;
 
     let cluster_identifiers = cluster_identifiers_from(engine_state, stack, &state, call, true)?;
 
     let guard = state.lock().unwrap();
 
+    let ctrl_c = engine_state.ctrlc.as_ref().unwrap().clone();
+    let statement: String = call.req(engine_state, stack, 0)?;
+
     let scope: Option<String> = call.get_flag(engine_state, stack, "scope")?;
-    let with_meta = call.has_flag(engine_state, stack, "with-meta")?;
 
     debug!("Running analytics query {}", &statement);
 
-    let mut results: Vec<Value> = vec![];
-    for identifier in cluster_identifiers {
+    let mut streams = StreamMap::new();
+    let rt = Arc::new(Runtime::new().unwrap());
+    for identifier in cluster_identifiers.clone() {
         let active_cluster = get_active_cluster(identifier.clone(), &guard, span)?;
         let bucket = call
             .get_flag(engine_state, stack, "bucket")?
             .or_else(|| active_cluster.active_bucket());
         let maybe_scope = bucket.and_then(|b| scope.clone().map(|s| (b, s)));
 
-        results.extend(do_analytics_query(
-            identifier.clone(),
+        let stream = send_analytics_query(
             active_cluster,
             maybe_scope,
-            &statement,
+            statement.clone(),
             ctrl_c.clone(),
             span,
-            with_meta,
-            true,
-        )?);
-    }
+            rt.clone(),
+        )?;
+
+        let json_stream = JsonRowStream::new(stream);
+        let mut json_streamer = RawJsonRowStreamer::new(json_stream, "".to_string());
+
+        rt.block_on(async {
+            // Read prelude and signature
+            json_streamer.read_prelude().await.unwrap();
+            json_streamer.read_row().await.unwrap();
+
+            // Read row containing `results: [`
+            json_streamer.read_row().await.unwrap();
+        });
 
-    Ok(Value::List {
-        vals: results,
-        internal_span: span,
+        streams.insert(identifier, json_streamer);
     }
-    .into_pipeline_data())
+
+    let result_stream = AnalyticsStream { streams, span, rt };
+
+    Ok(PipelineData::from(ListStream::new(
+        result_stream,
+        span,
+        Some(ctrl_c.clone()),
+    )))
 }
 
 pub fn send_analytics_query(
```

Comment:
Having send and do is really confusing, should we rename these to make intent clear?

Reply:
Good point, I've taken send out of do, and made do `read_analytics_response`